Fix for #9 (register token for registry)

groboclown opened the pull request:
This adds in the new command line argument, `-k` or `--registry-token`, to allow for specifying a registry token for login.

Also, added a debug logging for reporting a stack trace.  It was helpful for debugging :)

Tested locally against an ECR repo.  This has room for improvement, such as allowing explicit declaration of the token type ("Basic" or "Bearer" or something else; right now, it's hard-coded to "Basic").

groboclown committed:
Add support for basic token authentication, such as with an AWS ECR repository.

groboclown committed:
Add support for basic token authentication, such as with an AWS ECR repository.

groboclown committed:
Add example from README into the cli.

groboclown committed:
Fixed formatting issues with checkin.

groboclown committed:
Fixed formatting issues with checkin.

groboclown committed:
Add unit tests to cover new functionality.

groboclown committed:
Add unit tests to cover new functionality.

joelee2012 commented on claircli/cli.py:
```diff
@@ -143,6 +155,16 @@ def analyze_image(self):
             args.images = self.resolve_images(args.images)
+        for domain_token in args.domain_tokens:
+            domain_token_split = domain_token.split(':', 1)
+            if len(domain_token_split) != 2:
```
check if domain_token_split[1] is empty as well

joelee2012 commented on claircli/__version__.py:
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-__version__ = '1.1'
+__version__ = '1.1.1'
```
1.2

groboclown committed:
Fixes based on code review.

Version 1.2, because this is an enhancement, not a bug fix.

Additional checks on the command line interpretation to ensure valid input, plus unit tests for this.

joelee2012 commented:
Thanks and merge it now
